### PR TITLE
Update introspection documentation

### PR DESCRIPTION
**Pull request by ChaimaaeROUAI**
Fixing issue #11207

### CODE REVIEW

File: src/main/docs/guide/appendix/architecture/introspectionArch.adoc [hunk 1]
```diff
@@ -1,19 +1,18 @@
 The goal of <<introspection, Bean Introspections>> is to provide an alternative to reflection and the JDK's link:{jdkapi}/java.desktop/java/beans/Introspector.html[Introspector API] that is coupled to the `java.desktop` module in recent versions of Java.
 
-Many libraries in Java need to programmatically discover what methods represent properties of a class in some way and whilst the https://en.wikipedia.org/wiki/JavaBeans[JavaBeans specification] tried to establish a standard convention, the language itself has evolved to include other constructs like Records that represent properties as components.
+Many libraries in Java need to programmatically discover what methods represent properties of a class in some way and while the https://en.wikipedia.org/wiki/JavaBeans[JavaBeans specification] tried to establish a standard convention, the language itself has evolved to include other constructs like Records that represent properties as components.
```
Comment (sdelamo) [inline src/main/docs/guide/appendix/architecture/introspectionArch.adoc]:
```suggestion
Many libraries in Java need to programmatically discover what methods represent properties of a class in some way and whilst the https://en.wikipedia.org/wiki/JavaBeans[JavaBeans specification] tried to establish a standard convention, the language itself has evolved to include other constructs like Records that represent properties as components.
```

Comment (sdelamo) [reply src/main/docs/guide/appendix/architecture/introspectionArch.adoc]: I think `whilst` was correct.

File: src/main/docs/guide/appendix/architecture/introspectionArch.adoc [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
 
 In addition, other languages like Kotlin and Groovy have native support for class properties that need to be supported at the framework level.
 
 The api:inject.beans.visitor.IntrospectedTypeElementVisitor[] visits declarations of the ann:core.annotation.Introspected[] annotation on types and generates at compilation time implementations of api:core.beans.BeanIntrospection[] that are associated with each annotated type:
 
 image::arch/introspections.png[]
 
-This generation happens via the `io.micronaut.inject.beans.visitor.BeanIntrospectionWriter`, an internal class that uses the ASM bytecode generation library to generate two additional classes.
+This generation happens via the `io.micronaut.inject.beans.visitor.BeanIntrospectionWriter`, an internal class that uses the ASM bytecode generation library to generate an additional classes.
```
Comment (sdelamo) [inline src/main/docs/guide/appendix/architecture/introspectionArch.adoc]:
```suggestion
This generation happens via the `io.micronaut.inject.beans.visitor.BeanIntrospectionWriter`, an internal class that uses the ASM bytecode generation library to generate an additional class.
```

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: src/main/docs/guide/appendix/architecture/introspectionArch.adoc (hunk 1 above)
Comment (ChaimaaeROUAI) [reply src/main/docs/guide/appendix/architecture/introspectionArch.adoc]: ah OK I thought It was a typo

**Review by sdelamo — APPROVED** [review]

**Review by graemerocher — APPROVED** [review]

**Pull request merged**